sssd_test_framework/utils/authentication.py

```diff
@@ -707,72 +759,109 @@ def password_expired(self, username: str, password: str, new_password: str, host
         :type new_password: str
         :param hostname: The hostname to connect to.
         :type hostname: str
-        :return: True if authentication and password change was successful, False otherwise.
-        :rtype: bool
+        :return: Tuple containing [except return code, command exit code, stdout, stderr].
+        :rtype: Tuple[int, int, str, str]
         """
         result = self.host.conn.expect_nobody(
             rf"""
+            # Disable debug output
+            exp_internal 0
+
+            proc exitmsg {{ msg code }} {{
+                # Close spawned program, if we are in the prompt
+                catch close
+
+                # Wait for the exit code
+                lassign [wait] pid spawnid os_error_flag rc
+
+                puts ""
+                puts "expect result: $msg"
+                puts "expect exit code: $code"
+                puts "expect spawn exit code: $rc"
+                exit $code
+            }}
+
             # It takes some time to get authentication failure
             set timeout {DEFAULT_AUTHENTICATION_TIMEOUT}
             set prompt "\n.*\[#\$>\] $"
+            log_user 1
+            log_file /tmp/expect.log
 
             spawn ssh {self.opts} \
                 -o PreferredAuthentications=password \
                 -o NumberOfPasswordPrompts=1 \
-                -l "{username}" "{hostname}"
+                -l "{username}" localhost
```

Comment:
Why change from `{hostname}` to `localhost`?   And if we change here, should hostname be dropped as an option from the method definition?   Do we know if hostname is passed in for any existing tests?

Reply:
I think this was a bad rebase, let me check.

Reply:
Nope, I just missed it. Nice catch! Thank you.